### PR TITLE
Added basic script to make api-token for culler

### PR DESCRIPTION
**Pull request by mroman-redhat**
## Related Issues and Dependencies

…

## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

Adds the JupyterHub Idle Culler as an external service

## Description

This PR adds the idle culler as an external service and also exposes its api-token into a secret the the idle culler can read from.

### CODE REVIEW

File: .jupyter/jupyterhub_config.py [hunk 1]
```diff
@@ -25,6 +25,9 @@
     'KUBERNETES_SERVICE_PORT': os.environ['KUBERNETES_SERVICE_PORT'],
     'JUPYTERHUB_LOGIN_URL': None
 }
+
+idle_culler_api_token = str(uuid.uuid4())
```
Comment (vpavlin) [inline .jupyter/jupyterhub_config.py:29]: This cannot be re-generated on every restart of the JH as the culler container would not get the new value and API calls woulds start failing after restart.

Ideally implement 2 functions:
* `get_culler_secret`
* `set_culler_secret`

Call the `get_culler_secret` here and only if the secret does not exist, call `set_culler_secret`

And use the loaded or created secret in the services dict

Also, the name of the secret should be something like `jupyterhub-idle-culler` and the key in it should be `api_token`

**Review by 4n4nd — COMMENTED** [review]
/lgtm

File: .jupyter/jupyterhub_config.py [hunk 2]
```diff
@@ -20,14 +20,39 @@
...
+    secret = openshift.read_secret('jupyterhub-api-token')
+    if secret == {}:
+        return set_culler_secret()
+    else:
+        return secret['token']
+
+def set_culler_secret():
+
+    secret_data = str(uuid.uuid4())
+    openshift.write_secret('jupyterhub-api-token', {'token': secret_data})
```
Comment (vpavlin) [inline .jupyter/jupyterhub_config.py:39]: Chanfe the name to something related to idle-culler - e.g. `jupyterhub-idle-culler`

**Pull request merged**